src/index.js

```diff
@@ -8,110 +9,65 @@ class RetryChunkLoadPlugin {
-          var error = new Error();
-          function loadScript(src, retries) {
-
-            var script = document.createElement('script');
-            var retryAttempt = ${maxRetries} - retries + 1;
-            var retryAttemptString = '&retry-attempt=' + retryAttempt;
-            var onScriptComplete;
-            ${
-              jsonpScriptType
-                ? `script.type = ${JSON.stringify(jsonpScriptType)};`
-                : ''
-            }
-            script.charset = 'utf-8';
-            script.timeout = ${chunkLoadTimeout / 1000};
-            if (${mainTemplate.requireFn}.nc) {
-              script.setAttribute("nonce", ${mainTemplate.requireFn}.nc);
-            }
-            script.src = src;
-            ${crossOriginLoading ? crossOriginScript : ''}
-            onScriptComplete = function (event) {
-              // avoid mem leaks in IE.
-              script.onerror = script.onload = null;
-              clearTimeout(timeout);
-              var chunk = installedChunks[chunkId];
-              if (chunk !== 0) {
-                if (chunk) {
-                  if (retries === 0) {
-                    var errorType = event && (event.type === 'load' ? 'missing' : event.type);
-                    var realSrc = event && event.target && event.target.src;
-                    error.message = 'Loading chunk ' + chunkId + ' failed after ${maxRetries} retries.\\n(' + errorType + ': ' + realSrc + ')';
-                    error.name = 'ChunkLoadError';
-                    error.type = errorType;
-                    error.request = realSrc;${
-                      this.options.lastResortScript
-                        ? this.options.lastResortScript
-                        : ''
-                    }
-                    chunk[1](error);
-                    installedChunks[chunkId] = undefined;
-                  } else {
-                    var cacheBust = ${getCacheBustString()} + retryAttemptString;
-                    var retryScript = loadScript(jsonpScriptSrc(chunkId) + '?' + cacheBust, (retries-1));
-                    document.head.appendChild(retryScript);
-                  }
-                } else {
-                  installedChunks[chunkId] = undefined;
+      const maxRetryValueFromOptions = Number(this.options.maxRetries);
+      const maxRetries =
+        Number.isInteger(maxRetryValueFromOptions) &&
+        maxRetryValueFromOptions > 0
+          ? maxRetryValueFromOptions
+          : 1;
+      const getCacheBustString = () =>
+        this.options.cacheBust
+          ? `
+                  (${this.options.cacheBust})();
+                `
+          : '"cache-bust=true"';
+      mainTemplate.hooks.localVars.tap(
+        { name: pluginName, stage: 1 },
+        source => {
+          const script = `
+          var oldGetScript = ${RuntimeGlobals.getChunkScriptFilename};
+          var oldLoadScript = ${RuntimeGlobals.ensureChunk};
+          var queryMap = new Map();
```

Comment:
Is it ok to use a Map here? I'm just thinking about preserving compatibility for older ES5 browsers, but I know that IE11 does support Map's so maybe it's ok. Do you know if webpack emit's Map objects?

Reply:
I think it is configurable using browserlist

Reply:
Ah ok, let's play it safe then, as the chunk id's should always be strings I think it's ok for these to just be plain old javascript objects. Please can you refactor the 2 Map's just to be objects?

Reply:
The loading runtime uses Promises, which is not an ES5 feature. and as per [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) IE11 supports Map and has, set and get.

Reply:
its your call

Reply:
Alright let's keep it as is, if people write in about it breaking people on old browsers it's an easy enough fix